collab-service/src/redis_adapter/redis_pubsub_types.d.ts

```diff
@@ -0,0 +1,13 @@
+declare interface TunnelPubSub<T, V> {
+  registerEvent(
+    call: (res: V) => void,
+  ): Promise<void>;
+
+  push(request: T): Promise<void>;
```

Comment:
Tunnel should be seen as a pipe, without further processing. It only works on a single data type.